### PR TITLE
fix_31_validate_ops

### PR DESCRIPTION
**Pull request by shrajfr12**
1. Assuming that validation means the serviceDescripor associated with
the grpc server has the method in methoddesc array.
2.Expose the serviceInfo in Grafeas and set after it is registered.
3. Function that takes a method name for servicename and checks if it exists in the serviceInfo map
4. Invoke the function wherever reqd. 
5. Q. Should this validation be optional ? Via startup config ? 
6. Q. the var methodName may also be not reqd as the string can be directly passed, but is there
so can later on use reflection to derive operation name from the calling function. Is the calling
function in api always likely to match operation name  and is reflect call performant ?
7. Tested by running client-go samples test and not getting any "not found" errors.

Thanks for any feedback and clarifications and will follow up with suggested changes.

### CODE REVIEW

**Comment by shrajfr12** [comment]
tests in v1alpha1/impl_test.go do not yet have  the ServiceInfo initialized, so will update PR with the changes ....

**Comment by vtsao** [comment]
Hey @shrajfr12 thanks for your PR.

I think you misunderstood the issue, we don't need to check if the API method is implemented. What the issue was about is to check if an occurrence or note was created with an operation, we want to call [GetOperation()](https://github.com/grafeas/grafeas/blob/master/samples/server/go-server/api/server/v1alpha1/impl.go#L202) to validate the operation object exists in the database.

An [operation](https://github.com/googleapis/googleapis/blob/master/google/longrunning/operations.proto#L83) can be used to track a group of notes or occurrences created for a common reason. For example, all the occurrences created during some scan of a number of resources (like Docker images) might be assigned the same operation.

Hope that clarifies things, feel free to adjust your PR accordingly. Thanks.

**Comment by shrajfr12** [comment]
Thanks @vtsao for the clarification.  Much appreciated. Might have more  follow up questions later as  I revise the PR.

**Comment by shrajfr12** [comment]
Hi @vtsao , Currently operation is not tied to Note or Occurrence so to make the validation,  Note and Occurrence needs to be first enhanced with an extra field with the operation name. So during an operation if a note needs to be created, this field will be set. which can then be used for validation.   Is this  assumption ok ?

**Comment by vtsao** [comment]
Ah yes you're right we have to add an operation_name field on the Occurrence and Note message. Please feel free to add that. Thanks!

File: v1alpha1/proto/grafeas.swagger.json [hunk 1]
```diff
@@ -1320,6 +1320,10 @@
           "type": "string",
```
[the remark this answers is not in the document]
Comment (shrajfr12) [reply v1alpha1/proto/grafeas.swagger.json]: Hi @vtsao, Thanks for your comments. I just pushed the generated files and updated branch.

**Comment by vtsao** [comment]
Thanks for this!

**Pull request merged**